mantidproject/mantid
# Fix calling `GetEiMonDet` from Python when using workspace indices

When using 'WorkspaceIndex' as _IndexType_ in `GetEiMonDet`, the algorithm casts the _Monitor_ property directly to a `size_t`. This does not work when the algorithm is called from Python since `size_t` is unsigned and Python can by default only pass signed integers. This PR now fixes the problem and also introduces a new sanity check (_Monitor_ cannot be negative if it is a workspace index).

**To test:**

Running

``` python
import numpy

x = numpy.tile(numpy.arange(-10.0, 10.1, 0.5), 2)
spectrum = numpy.zeros(len(x) - 1)
spectrum[len(spectrum) / 2 - 2: len(spectrum) /  2 + 2] = 1 # Fake elastic peak
y = numpy.tile(spectrum, 2)
ws = CreateWorkspace(x, y, NSpec=2, UnitX='TOF')
EditInstrumentGeometry(Workspace=ws,
                       PrimaryFlightPath=0.01,
                       L2=[0.01, 0.1],
                       Polar=[180, 0],
                       InstrumentName='Omnium')

eppTable = FindEPP(InputWorkspace=ws)
GetEiMonDet(DetectorWorkspace=ws,
            DetectorEPPTable=eppTable,
            IndexType='WorkspaceIndex',
            Detectors='1',
            Monitor='0',
            PulseInterval=22.0,
            NominalIncidentEnergy=200.0)
```

before applying these changes will produce an error. After applying the changes, the above script should run without problems.

Fixes #17623 

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
